smoke-tests/spec/external_dns_helper.rb

```diff
@@ -78,21 +65,28 @@ def delete_txt_record(zone_id, domain_name, namespace)
 
 # Checks if the zone is empty, then deletes
 # if not empty, it will assume it contains one A record and one TXT record created by external-dns
-def cleanup_zone(zone_id, domain, namespace, ingress_name)
-  if is_zone_empty?(zone_id)
-    # delete_zone(zone_id)
-  else
-    delete_a_record(zone_id, domain, namespace, ingress_name)
-    delete_txt_record(zone_id, domain, namespace)
+def cleanup_zone(domain, namespace, ingress_name, zone_id = nil)
+  if zone_id.nil?
+    zone_id = get_zone_by_name(domain)
+  end
+
+  records = get_zone_records(zone_id)
+
+  unless is_zone_empty?(records)
+    records.each do |record|
+      case record[:type]
```

Comment:
Do you need different methods to delete TXT versus A records? If you're passing the record object into the delete function, can you replace "delete_a_record" and "delete_txt_record" with a single "delete_record" method?

Reply:
you are right, it is much simpler that way.